Tests fails on nanosp physical device assumes navigation possible on all backends

When running ragger testing on a physical device the most basic test fails. All other tests pass. All tests pass on the speculos emulator. 

Steps to reproduce, using nanosp device with app version 1.4.4 and firmware 1.0.4, includes [PR #14](https://github.com/LedgerHQ/app-eos/pull/14)

1. Install app on device
```
sudo docker run --rm -ti -v "/dev/bus/usb:/dev/bus/usb" -v "$(realpath .):/app" --privileged ledger-app-builder:latest
make clean
BOLOS_SDK=$NANOSP_SDK make
BOLOS_SDK=$NANOSP_SDK make load
```
Follow prompts on device to remove previous version and install new version.
2. Open eos app on device
3. Run test
```
cd tests/functional
pytest -v --tb=short --nanosp --backend ledgerwallet -k test_app_mainmenu_settings
```

### Error Report
==================================================== test session starts ====================================================
platform linux -- Python 3.10.6, pytest-7.2.0, pluggy-1.0.0 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/eric/repos/ENF/ledger-app/tests/functional
collected 18 items / 17 deselected / 1 selected                                                                             

test_app_mainmenu_settings_cfg.py::test_app_mainmenu_settings_cfg[nanosp 1.0.3] FAILED                                [100%]

========================================================= FAILURES ==========================================================
_______________________________________ test_app_mainmenu_settings_cfg[nanosp 1.0.3] ________________________________________
test_app_mainmenu_settings_cfg.py:37: in test_app_mainmenu_settings_cfg
    assert data_allowed is True
E   assert False is True
--------------------------------------------------- Captured stderr setup ---------------------------------------------------
[2023-01-05 05:29:46,759][INFO] ragger.logger - Starting LedgerWalletBackend stream
---------------------------------------------------- Captured log setup -----------------------------------------------------
INFO     ragger.logger:ledgerwallet.py:54 Starting LedgerWalletBackend stream
--------------------------------------------------- Captured stderr call ----------------------------------------------------
[2023-01-05 05:29:46,837][DEBUG] ragger.apdu_logger - => d406000000
[2023-01-05 05:29:46,842][DEBUG] ragger.apdu_logger - <= 000104049000
[2023-01-05 05:29:46,842][DEBUG] ragger.apdu_logger - <= 000104049000
[2023-01-05 05:29:46,843][DEBUG] ragger.apdu_logger - => d406000000
[2023-01-05 05:29:46,848][DEBUG] ragger.apdu_logger - <= 000104049000
[2023-01-05 05:29:46,849][DEBUG] ragger.apdu_logger - <= 000104049000
----------------------------------------------------- Captured log call -----------------------------------------------------
DEBUG    ragger.apdu_logger:ledgerwallet.py:82 => d406000000
DEBUG    ragger.apdu_logger:ledgerwallet.py:86 <= 000104049000
DEBUG    ragger.apdu_logger:ledgerwallet.py:37 <= 000104049000
DEBUG    ragger.apdu_logger:ledgerwallet.py:82 => d406000000
DEBUG    ragger.apdu_logger:ledgerwallet.py:86 <= 000104049000
DEBUG    ragger.apdu_logger:ledgerwallet.py:37 <= 000104049000
================================================== short test summary info ==================================================
FAILED test_app_mainmenu_settings_cfg.py::test_app_mainmenu_settings_cfg[nanosp 1.0.3] - assert False is True
============================================= 1 failed, 17 deselected in 2.16s ==============================================

--- a/tests/functional/test_app_mainmenu_settings_cfg.py
+++ b/tests/functional/test_app_mainmenu_settings_cfg.py
@@ -1,3 +1,4 @@
+from ragger.backend import SpeculosBackend
 from ragger.navigator import NavInsID, NavIns
 
 from apps.eos import EosClient
@@ -13,26 +14,32 @@ def test_app_mainmenu_settings_cfg(backend, navigator, test_name):
     client = EosClient(backend)
 
     # Get appversion and "data_allowed parameter"
+    # This works on both the emulator and a physical device
     data_allowed, version = client.send_get_app_configuration()
     assert data_allowed is False
     assert version == (MAJOR, MINOR, PATCH)
 
-    # Navigate in the main menu and the setting menu
-    # Change the "data_allowed parameter" value
-    instructions = [
-        NavIns(NavInsID.RIGHT_CLICK),
-        NavIns(NavInsID.RIGHT_CLICK),
-        NavIns(NavInsID.RIGHT_CLICK),
-        NavIns(NavInsID.LEFT_CLICK),
-        NavIns(NavInsID.BOTH_CLICK),
-        NavIns(NavInsID.BOTH_CLICK),
-        NavIns(NavInsID.RIGHT_CLICK),
-        NavIns(NavInsID.BOTH_CLICK)
-    ]
-    navigator.navigate_and_compare(ROOT_SCREENSHOT_PATH, test_name, instructions,
+    # scoping navigation and next test to the emulator
+    # navigation instructions are not applied to physical devices
+    # without navigation instructions allow data remmained unchanged
+    #    no sense in running this test when there is no change
+    if isinstance(backend, SpeculosBackend):
+        # Navigate in the main menu and the setting menu
+        # Change the "data_allowed parameter" value
+        instructions = [
+            NavIns(NavInsID.RIGHT_CLICK),
+            NavIns(NavInsID.RIGHT_CLICK),
+            NavIns(NavInsID.RIGHT_CLICK),
+            NavIns(NavInsID.LEFT_CLICK),
+            NavIns(NavInsID.BOTH_CLICK),
+            NavIns(NavInsID.BOTH_CLICK),
+            NavIns(NavInsID.RIGHT_CLICK),
+            NavIns(NavInsID.BOTH_CLICK)
+        ]
+        navigator.navigate_and_compare(ROOT_SCREENSHOT_PATH, test_name, instructions,
                                    screen_change_before_first_instruction=False)
 
-    # Check that "data_allowed parameter" changed
-    data_allowed, version = client.send_get_app_configuration()
-    assert data_allowed is True
-    assert version == (MAJOR, MINOR, PATCH)
+        # Check that "data_allowed parameter" changed
+        data_allowed, version = client.send_get_app_configuration()
+        assert data_allowed is True
+        assert version == (MAJOR, MINOR, PATCH)

--- a/tests/functional/test_app_mainmenu_settings_cfg.py
+++ b/tests/functional/test_app_mainmenu_settings_cfg.py
@@ -21,7 +21,7 @@ def test_app_mainmenu_settings_cfg(backend, navigator, test_name):
 
     # scoping navigation and next test to the emulator
     # navigation instructions are not applied to physical devices
-    # without navigation instructions allow data remmained unchanged
+    # without navigation instructions allow data remained unchanged
     #    no sense in running this test when there is no change
     if isinstance(backend, SpeculosBackend):
         # Navigate in the main menu and the setting menu

